src/main/java/org/jabref/gui/importer/GenerateEntryFromIdAction.java

```diff
@@ -43,7 +44,17 @@ public void execute() {
         backgroundTask.titleProperty().set(Localization.lang("Import by ID"));
         backgroundTask.showToUser(true);
         backgroundTask.onRunning(() -> dialogService.notify("%s".formatted(backgroundTask.messageProperty().get())));
-        backgroundTask.onFailure((e) -> dialogService.notify(e.getMessage()));
+        backgroundTask.onFailure((e) -> {
+            // When unable to import by ID, present the user options to cancel or add entry manually
+            boolean addEntryFlag = dialogService.showConfirmationDialogAndWait(Localization.lang("Failed to import by ID"),
+                                    e.getMessage(),
+                                   Localization.lang("Add entry manually"));
+            if (addEntryFlag) {
+                // add entry manually
+                new NewEntryAction(libraryTab.frame(), StandardEntryType.Article, dialogService,
+                                    preferencesService, libraryTab.stateManager()).execute();
```

Comment:
Can you please inject the stateManager in the constructor (just like PreferencesService)?

Reply:
@calixtus , thanks for the feedback. I created a new stateManager in that class